traversability_estimation/src/TraversabilityMap.cpp

```diff
@@ -491,20 +496,27 @@ bool TraversabilityMap::isTraversable(grid_map::Polygon& polygon, double& traver
 {
   unsigned int nCells = 0;
   traversability = 0.0;
+  bool isSafe = true;
 
   // Iterate through polygon and check for traversability.
   for (grid_map::PolygonIterator polygonIterator(traversabilityMap_, polygon);
       !polygonIterator.isPastEnd(); ++polygonIterator) {
 
     // Check for slopes
-    if (!checkForSlope(*polygonIterator)) return false;
+    if (!checkForSlope(*polygonIterator)) {
```

Comment:
The idea of returning as soon as one check is false is to avoid the overhead of evaluating all checks although you know that the polygon is not traversable. Why do you want to do all the checks? Additionally the traversability value was set to zero if one check fails, which is not the case anymore.